scripts/o3de/o3de/packaging.py

```diff
@@ -0,0 +1,318 @@
+    #             print("Non-monolithic projects not supported!")
+    #             return 1
+    # else:
+    #     return package_none()
+    external_script_path = args.script_file #"C:\\workspace\\projects\\NewspaperDeliveryGame\\export_rules\\test_export.py"
+    module_name = os.path.basename(external_script_path)
+
+    #works for python 3.5+
+    sys.path.insert(0, os.path.split(pathlib.Path(external_script_path))[0])
+    sys.path.insert(0, os.getcwd())
+    spec = importlib.util.spec_from_file_location(module_name, external_script_path)
+    foo = importlib.util.module_from_spec(spec)
+    sys.modules[module_name] = foo
+    foo.o3de_project_path =  args.project_path #customer facing variable name for accessible scripting
+    spec.loader.exec_module(foo)
+    return 0
+
+
+
+def add_parser_args(parser):
+    #if no use case is specified, we assume standalone
+    use_case_group = parser.add_mutually_exclusive_group(required=False)
+    # use_case_group.add_argument('-s', '--standalone', action='store_true',
+    #                    default=True,
+    #                    help='builds as standalone.')
+    
+    #need to get terminology termed out - i.e. packaging.py can be extremely confusing.
+    #need to get dictionary standardized, check with @stankowi on this, need steps and terms fully defined
+
+    #no need to future proof if there's a good off ramp. If our code is easily modifiable, the customer can handle the edge cases
+    #for iterative builds, make sure they are not trapped in a dead end. Need Asset bundling addressed
+
+    #something to also think about: what would a long term GUI look like for this?
+    #GUI can output config files and commands issued in packaging/build process. Robust code echo mode like in Blender
+
+    #try to sit down with carbonated and cyanide for their build processes. They need thrad obfuscation and boundaries, anti-cheat, modding, etc.
+
+    #simulation customers: two challenges
+    #1) live-feed/simulation and human interface use cases. deploying to dedicated enviroments
+    #2) they don't need as much as the game teams, but it could conflict with the pipeline.
+
+    #
+
+    # parser.add_argument('-m', '--monolithic', action='store_true',
+    #                    default=False,
+    #                    help='bundles as monolithic game release.')
+    parser.add_argument('-sf', '--script-file', type=pathlib.Path, required=True, help='external script file to run')
+    # parser.add_argument('-z', '--zipped', action='store_true',
+    #                     default=False,
+    #                     help='takes the packaged build and zips it')
+    parser.add_argument('-pp', '--project-path', type=pathlib.Path, required=False,
+                        help='Project to package')
+    # parser.add_argument('-ep', '--engine-path', type=pathlib.Path, required=False,
+    #                     help='Engine for building tooling')
+    parser.set_defaults(func=_run_packaging)
+    
```

Comment:
in this prototype, we only take 2 arguments: project path, and script file path. These are separate because we have no guarantees as to where the user put the script file.